src/Manager.sol

```diff
@@ -405,79 +401,78 @@ abstract contract Manager is
+        }
 
         if (amount == 0) {
             revert ZeroAmount();
         }
 
-        if (shouldCheckRateLimit) {
-            // Lock/burn tokens before checking rate limits
-            if (_mode == Mode.LOCKING) {
-                // use transferFrom to pull tokens from the user and lock them
-                // query own token balance before transfer
-                uint256 balanceBefore = getTokenBalanceOf(_token, address(this));
+        // Lock/burn tokens before checking rate limits
+        if (_mode == Mode.LOCKING) {
+            // use transferFrom to pull tokens from the user and lock them
+            // query own token balance before transfer
+            uint256 balanceBefore = getTokenBalanceOf(_token, address(this));
 
-                // transfer tokens
-                IERC20(_token).safeTransferFrom(msg.sender, address(this), amount);
+            // transfer tokens
+            IERC20(_token).safeTransferFrom(msg.sender, address(this), amount);
 
-                // query own token balance after transfer
-                uint256 balanceAfter = getTokenBalanceOf(_token, address(this));
+            // query own token balance after transfer
+            uint256 balanceAfter = getTokenBalanceOf(_token, address(this));
 
-                // correct amount for potential transfer fees
-                amount = balanceAfter - balanceBefore;
-            } else if (_mode == Mode.BURNING) {
-                // query sender's token balance before transfer
-                uint256 balanceBefore = getTokenBalanceOf(_token, msg.sender);
+            // correct amount for potential transfer fees
+            amount = balanceAfter - balanceBefore;
+        } else if (_mode == Mode.BURNING) {
+            // query sender's token balance before transfer
+            uint256 balanceBefore = getTokenBalanceOf(_token, msg.sender);
 
-                // call the token's burn function to burn the sender's token
-                ERC20Burnable(_token).burnFrom(msg.sender, amount);
+            // call the token's burn function to burn the sender's token
+            ERC20Burnable(_token).burnFrom(msg.sender, amount);
 
-                // query sender's token balance after transfer
-                uint256 balanceAfter = getTokenBalanceOf(_token, msg.sender);
+            // query sender's token balance after transfer
+            uint256 balanceAfter = getTokenBalanceOf(_token, msg.sender);
 
-                // correct amount for potential burn fees
-                amount = balanceAfter - balanceBefore;
-            } else {
-                revert InvalidMode(uint8(_mode));
-            }
+            // correct amount for potential burn fees
+            amount = balanceAfter - balanceBefore;
+        } else {
+            revert InvalidMode(uint8(_mode));
+        }
+
+        // now check rate limits
+        bool isAmountRateLimited = _isOutboundAmountRateLimited(amount);
+        if (shouldQueue && isAmountRateLimited) {
```

Comment:
Looks like we've lost the revert if `shouldQueue = false`?

Reply:
The revert is handled in `_consumeOutboundAmount`

Reply:
Ok nice I didn't see it in this diff and was worried we'd lost that path